File: oblig1/RungeKutta.m
function [ x ] = RungeKutta( b, y, f, h)
% Embedded Runge Kutta

m=length(y);
x=zeros(m,1);



k1 = f(y);
k2 = f(y+h*((1/3)*k1));
k3 = f(y+h*((1/6)*k1 + (1/6)*k2));
k4 = f(y+h*((1/8)*k1 + (3/8)*k3));
k5 = f(y+h*((1/2)*k1 + (-3/2)*k3 + 2*k4));

x = y + h*(b(1)*k1 + b(2)*k2 + ...
           b(3)*k3 + b(4)*k4 + b(5)*k5);



end

File: week3/runge_kutta.py
"""
implement the Runge-Kutta method.
input vector c, b and A, with yn and f
"""
import numpy as np
def runge_kutta(b):
    """
    usage
    """
    mu = len(b)
    #compute ksi
    """
    for n in range(0, N):
        for m in range(0, M):
            for i in range(1,mu):
                ksi[m][i] = y[m][n] + h*sum(a[r][i]*f(t[n]+c[i]*h,ksi[m][i]))
    
    for n in range(0, N):
        for m in range(0,M):
            for j in range(1,mu+1)
                y[m][n+1] = y[m][n] + h*sum(b[j]*f(t[n]+c[j]*h,ksi[j]))[m][n]
    return y

    """
    print b
    
b = np.array([1./6,1/3.,1/3.,1/6.])
runge_kutta(b)